dropwizard-client/src/test/java/io/dropwizard/client/DropwizardApacheConnectorTest.java

```diff
@@ -0,0 +1,180 @@
+        Client client = createStandardClient();
+
+        // before override
+        assertThatGetConnectonTimeoutForTarget(
+                client.target(URI_THAT_CANNOT_BE_CONNECTED_TO)
+        ).isLessThan(DEFAULT_CONNECT_TIMEOUT + 300);
+
+        // after override
+        assertThatGetConnectonTimeoutForTarget(
+                client.target(URI_THAT_CANNOT_BE_CONNECTED_TO)
+                        .property(ClientProperties.CONNECT_TIMEOUT, DEFAULT_CONNECT_TIMEOUT + 400)
+        ).isGreaterThan(DEFAULT_CONNECT_TIMEOUT + 400);
+    }
+
+    @Test
+    public void when_no_override_then_redirected_request_successfully_redirected() {
+        Client client = createStandardClient();
+
+        assertThat(
+                client.target(TEST_URI + "/redirect")
+                        .request()
+                        .get(String.class)
+        ).isEqualTo("redirected");
+    }
+
+    @Test
+    public void when_configuration_overridden_to_disallow_redirects_temporary_redirect_status_returned() {
+        Client client = createStandardClient();
+
+        assertThat(
+                client.target(TEST_URI + "/redirect")
+                        .property(ClientProperties.FOLLOW_REDIRECTS, false)
+                        .request()
+                        .get(Response.class)
+                        .getStatus()
+        ).isEqualTo(HttpStatus.SC_TEMPORARY_REDIRECT);
+    }
+
+    @Path("/")
+    public static class TestResource {
+
+        @GET
+        @Path("/long_running")
+        public String getWithSleep() throws InterruptedException {
+            TimeUnit.MILLISECONDS.sleep(SLEEP_TIME_IN_MILLIS);
+            return "success";
+        }
+
+        @GET
+        @Path("redirect")
+        public Response getWithRedirect() {
+            return Response.temporaryRedirect(URI.create("/redirected")).build();
+        }
+
+        @GET
+        @Path("redirected")
+        public String redirectedGet() {
+            return "redirected";
+        }
+
+    }
+
+    public static class TestApplication extends Application<Configuration> {
+        public static void main(String[] args) throws Exception {
+            new TestApplication().run(args);
+        }
+
+        @Override
+        public void run(Configuration configuration, Environment environment) throws Exception {
+            environment.jersey().register(TestResource.class);
+        }
+    }
+
+    private static AbstractLongAssert<?> assertThatGetConnectonTimeoutForTarget(WebTarget webTarget) {
+        final long startTime = System.currentTimeMillis();
+        ProcessingException processingException = null;
+        try {
+            webTarget
+                    .request()
+                    .get(Response.class);
+        } catch (ProcessingException e) {
+            processingException = e;
+        }
+        final long endTime = System.currentTimeMillis();
```

Comment:
I think we could move this logic to the exception block and throw an `AssertionError` if an excption has not happened. Something like this:
````
final long startTime = System.currentTimeMillis();
try {
    webTarget
            .request()
            .get(Response.class);
    throw new AssertionError("Should fail with ProcessingException");
} catch (ProcessingException e) {
    final long endTime = System.currentTimeMillis();
    assertThat(e).isNotNull();
    //noinspection ConstantConditions
    assertThat(e.getCause()).isNotNull();
    assertThat(e.getCause()).isInstanceOf(ConnectTimeoutException.class);
    return assertThat(endTime - startTime);
}
````